lib/tasks/sumofus.rake

```diff
@@ -2,50 +2,146 @@ require 'open-uri'
+        page.liquid_layout_id = layout_id
+        page.language_id = language_ids[locale]
+        page.content = I18n.t('fundraiser.generic.body', locale: locale)
+        page.save!
+        page.images.create!(content: image_handle) if page.images.empty?
+        pages[locale] = page
+      end
+      pages
+    end
 
-    image_handle = open(ARGV[2])
+    def manage_newlines(content)
+      content.
+        gsub(/(?:\n\r?|\r\n?)/, '<br>').
+        gsub(/<br *\/*>/, '<br>').
+        gsub(/<div><br>\t&nbsp;<\/div>/, '<br>').
+        gsub(/(<br>)*\s*<\/p>\s*(<br>)*\s*<p>\s*(<br>)*/, '</p><br><p>').
+        gsub(/(<br>)*\s*<\/div>\s*(<br>)*\s*<div>\s*(<br>)*/, '</p><br><p>').
+        gsub(/(\s*<br>\s*){3,}/, '<br><br>')
+    end
 
-    parsed_data = JSON.load data
-    count = 0
-    layout_id = LiquidLayout.where(title: 'Petition With Small Image').first.id
-    parsed_data.each_pair do |k, entry|
-      count +=1
+    def language_ids
+      return @language_ids unless @language_ids.blank?
+      @language_ids = {}
+      Language.all.each do |l|
+        @language_ids[l.code] = l.id
+      end
+      @language_ids
+    end
+
+    def clean_title(entry)
       title = entry['title'].blank? ? entry['petition_ask'] : entry['title']
-      page = Page.find_or_create_by!(title: title, liquid_layout_id: layout_id)
-      page.content = entry['page_content'].gsub(/(?:\n\r?|\r\n?)/, '<br>')
-      Page.reset_counters(page.id, :actions)
-      Page.update_counters(page.id, action_count: entry['signature_count'])
+      title.chomp(' ').chomp('!').chomp('.')
+    end
+
+    def duplicate_titles(titles)
+      titles.map do |title, variations|
+        variations.size > 1 ? title : nil
+      end.compact
+    end
+
+    def unique_titles(page_data)
+      titles = Hash.new({})
+      # byebug
+      page_data.each_pair do |k, entry|
+        title = clean_title(entry)
+        slug = entry['slug']
+        case titles[title].size
+        when 1
+          titles[title][slug] = "#{title} now"
+        when 2
+          titles[title][slug] = "#{title} now!"
+        when 3
+          titles[title][slug] = "#{title} today"
+        else # includes 0 case
+          titles[title] = {}
+          titles[title][slug] = title
+        end
+      end
+      titles
+    end
+
+    pages_before = Page.count
+    start_timestamp = Time.now
+
+    count, existing_image = 0, nil
+    petition_layout_id = LiquidLayout.where(title: 'Petition With Small Image').first.id
+    fundraiser_layout_id = LiquidLayout.where(title: 'Fundraiser With Large Image').first.id
+
+    post_action_pages = create_post_action_pages(fundraiser_layout_id, follow_image_handle)
+    titles = unique_titles(page_data)
+
+    duplicate_titles(titles).each do |title|
+      Page.where(title: title).each do |p|
+        p.update_attributes(title: titles[title][p.slug])
+      end
+    end
+
+    page_data.each_pair do |k, entry|
```

Comment:
This block here is pretty massive but since this isn't something we'll need to maintain I think it's fine